internal/stacksapi/stack_notification.go

```diff
@@ -0,0 +1,36 @@
+package stacksapi
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+)
+
+type CreateStackNotificationRequest struct {
+	StackKey string `json:"-"`      // The key to call the stack. Required.
+	JobUUID  string `json:"-"`      // The uuid of a job
+	Detail   string `json:"detail"` // Short notification message (max length 255)
+}
+
+func (c *Client) CreateStackNotification(ctx context.Context, req CreateStackNotificationRequest, opts ...RequestOption) (http.Header, error) {
+	const croppedMessage = "… (cropped)"
+	const maxDetailSize = 255 - len(croppedMessage)
```

Comment:
[non-blocking] this message takes up a fairly significant portion of the space in the notification detail (11/255 ≈ 5%). i wonder if it makes more sense to return an error early if the size exceeds the limit rather than cropping

Reply:
IMO 255 is a quite an arbitrary number. Our main objective is to ensure that the message is short, whether the limit is 255 or 244 doesn't matter very much, so I guess it's okay to effectively make the limit 244. 

I am against returning error here because I think it's better to get some signal than no signal. Also the caller side probably can't do much about the error apart from logging it, in such case, we already handle that for the caller on this layer.